src/rpc/rpcquorums.cpp

```diff
@@ -186,6 +187,16 @@ UniValue quorum_dkgstatus(const JSONRPCRequest& request)
         if (fMasternodeMode) {
             const CBlockIndex* pindexQuorum = chainActive[tipHeight - (tipHeight % params.dkgInterval)];
             auto expectedConnections = llmq::CLLMQUtils::GetQuorumConnections(params.type, pindexQuorum, activeMasternodeInfo.proTxHash);
+            if (sporkManager.IsSporkActive(SPORK_21_QUORUM_ALL_CONNECTED)) {
+                // when SPORK_21_QUORUM_ALL_CONNECTED is active, GetQuorumConnections only returns the connections
+                // that we should initiate but not the ones that should come in.
```

Comment:
Maybe add smth like `bool fOutboundOnly` to `GetQuorumConnections` params, tweak conditions accordingly `if (!fOutboundOnly || dmn->proTxHash < forMember) {` and then pass `false` here and `true` in `EnsureQuorumConnections`?

Reply:
Sounds good, will do this

Reply:
Done, with a small difference. I decided to unconditionally return all connections when `!onlyOutbound`, even if spork21 is not active.